connections/multiselect2.md

```diff
@@ -38,11 +38,14 @@ This section only applies if Multiselect 2 is run over a transport that is not n
 
 Some handshake protocols (TLS 1.3, Noise) support sending of *Early Data*. 
 
-In Multiselect 2 the server makes use of Early Data by sending a list of stream multiplexers. This ensures that the client can choose a stream multiplexer as soon as the handshake completes (or fail the connection if it doesn't support any stream multiplexer offered by the server).
+In Multiselect 2 endpoints make use of Early Data to speed up stream multiplexcer selection. As soon as an endpoints reaches a state during the handshake where it can send encrypted application data, it sends a list of supported stream multiplexers. The list SHOULD be ordered be ordered by descending preference.
 
-When using TLS 1.3, the server can send Early Data after it receives the ClientHello. Early Data is encrypted, but at this point of the handshake the client's identity is not yet verified.
+When using TLS 1.3, the server can send Early Data after it receives the ClientHello. Early Data is encrypted, but at this point of the handshake the client's identity is not yet verified. 
 While Noise in principle allows sending of unencrypted data, endpoints MUST NOT use this to send their list of stream multiplexers. An endpoint MAY send it as soon it is possible to send encrypted data, even if the peers' identity is not verified at that point.
 
+The stream multiplexer that is used in the connection is determined by intersection the lists sent by both endpoints, as follows: First all stream multiplexers that aren't supported by both endpoints are removed from the clients' list of stream multiplexers. The stream multiplexer chosen is then the first element of this list.
```

Comment:
What's the ordering for intersection?
What happens if one side sends `a,b,c` and the other side sends `b,c,a`?

Reply:
Depends on who sent what:

Client: `a,b,c`
Server: `b,c,a`
Remove all unsupported muxers from the clients list: `a,b,c`
Take the first: `a`

Client: `b,c,a`
Server: `a,b,c`
Remove all unsupported muxers from the clients list: `b,c,a`
Take the first: `b`

Reply:
This also means that only the ordering of the client's list matters. This is by design. In most of our protocols, the initiator gets to pick if there are multiple options.
If there's a good reason, of course we can turn this around easily.

Reply:
That's fine but it should be specified that the client list takes precedence.

Reply:
@vyzo Done.